ykcapi/src/lib.rs

```diff
@@ -178,9 +228,9 @@ pub extern "C" fn yk_stopgap(
     // Restore saved registers from the stack.
     let registers = Registers::from_ptr(rsp);
 
-    let mut sginterp = unsafe { SGInterp::new(activeframes, retvalptr) };
+    let mut sginterp = unsafe { SGInterp::new(activeframes, frameaddr) };
```

Comment:
For some reason I was expecting yksgi to disappear. Are we keeping it?

Reply:
All the initialisation code from SGI is still needed for stack reconstruction as well. I'm also a bit hesitant to remove SGI entirely as it might be useful in the future for debugging purposes. I imagined that at some point we have a switch to toggle stack reconstruction/stopgap interpreter.

Reply:
I find it confusing that we instantiate a stop gap interpreter, but then don't *really* use it.

Reply:
I tend to agree with Edd. If we do ever need a stopgap interpreter, we'll probably have to rewrite more-or-less from scratch anyway. I'd be inclined to remove it entirely.

Reply:
There's always the git history.

Reply:
Agreed. I'm going to resolve this particular comment but let's assume that we rip out most of the SGI in an upcoming PR.